configure

```diff
@@ -808,10 +810,34 @@ $opts{'boost'} = '' if exists $opts{'boost-version'} && !exists $opts{'boost'};
 my %use_system_pkg = map {$_, 1} qw/ant glib qt boost/;
 my %use_win_default = ('boost' => 'c:\\Boost');
 
+
+sub default_java_home {
+  my $host = $opts{'host'};
+  my $result = '';
+  if ($host eq 'macosx') {
+     $result = `/usr/libexec/java_home`;
+  }
+  elsif ($host ne 'win32') {
+     $result = `readlink -f /usr/bin/javac | sed "s:/bin/javac::"`;
```

Comment:
javac (or the user's preferred javac) may not be in /usr/bin, you can use the existing function "which()" to see if javac is on the PATH